### PR TITLE
feat: add VisdomLogger class for PyTorch training logs

### PR DESCRIPTION
**Pull request by Jayantparashar10**
## Description
Adds `visdom/pytorch.py` with `VisdomLogger`, a context manager that wraps `viz.line()` for training loops. Users call `tracker.log(name, value)` per metric the logger handles window creation, step counting, and `log_every `throttling internally.

## Motivation and Context
Logging a single metric with the raw Visdom API requires tracking the window handle, the current step, and whether to call `line()` with `update="append"` or not. `VisdomLogger` keeps that state internally so the training loop stays focused on training.

## How Has This Been Tested?
Ran `example/train_example.py` end-to-end against a live Visdom server with a 4-class MLP trained for 50 epochs on synthetic data. All five metrics — Train Loss, Train Accuracy, Learning Rate, Val Loss, Val Accuracy — rendered correctly with epoch 1–50 on the x-axis and consistent axis labels across every window.

Verified:
  - x-axis reads "epoch" and goes 1–50 on all graphs with no manual x= argument in the training loop
  - Train Loss curve decreases as expected; Learning Rate shows the correct staircase decay from StepLR
  - Val Loss and Val Accuracy track correctly per epoch
  - No stale window data from previous runs

## Screenshots (if appropriate):

Screenshot 2026-07-05 at 9 53 25 PM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactor or cleanup (changes to existing code for improved readability or performance)

## Checklist:

- [ ] I adapted the version number under `py/visdom/VERSION` according to [Semantic Versioning](https://semver.org/)
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: py/visdom/pytorch.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        else:
+            self.viz.line(
+                X=[step],
+                Y=[value],
+                win=self._wins[name],
+                env=self.env,
+                update="append",
+            )
+
+        self._step[name] = step + 1
```
[the remark this answers is not in the document]
Comment (Jayantparashar10) [reply py/visdom/pytorch.py]: Tensor values are handled via `hasattr(value, "item")`. On `np.asarray` — Visdom line() internally converts lists through `_ndarray()`, so wrapping with numpy is not needed here.

**Review by Manik-Khajuria-5 — CHANGES_REQUESTED** [review]
@Jayantparashar10 I read ur backend code changes most of the implementation is good Just handle type check in name and value properly and in append call please add opts also (Nice work!)

File: py/visdom/pytorch.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        self._step = {}
+        self._counter = {}
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        return False
+
+    def log(self, name, value, x=None, xlabel="epoch"):
```
Comment (tonypzy) [inline py/visdom/pytorch.py:72]: can you check the validity of name and value

**Review by tonypzy — CHANGES_REQUESTED** [review]
please resolve comments

**Review by tonypzy — APPROVED** [review]

**Review by Manik-Khajuria-5 — APPROVED** [review]

**Review by rajnisht7 — APPROVED** [review]
